### ISSUE TITLE
Compiling with clang compilers on Ubuntu runners causes OpenMP linking issues

### ISSUE THREAD

**Issue by IAlibay** [MDAnalysis/mdanalysis#3176]
## Expected behavior ##

Compiling by setting CC and CXX appropriately to capture the clang compilers should work on both MacOS and Ubuntu.

## Actual behavior ##

I'm in the midst of setting up a gh action cron job to do a compiler sweep for both clang and gcc, Turns out that if I attempt to set CC=clang and CXX=clang++ and then install MDAnalysis via setup I get OpenMP linking issues. It's not immediately clear to me if the issue is down to the gh action runners or generally an issue with Ubuntu.

Here is an example failed job: https://github.com/IAlibay/mdanalysis/runs/2109402006?check_suite_focus=true

**Comment by hmacdope** [comment]
Looks like it might need `-fopenmp`.

**Comment by hmacdope** [comment]
Some things appear to be being compiled with gcc.

`gcc -pthread -shared -B /usr/share/miniconda/envs/test/compiler_compat -L/usr/share/miniconda/envs/test/lib -Wl,-rpath=/usr/share/miniconda/envs/test/lib -Wl,--no-as-needed -Wl,--sysroot=/ build/temp.linux-x86_64-3.8/home/runner/work/mdanalysis/mdanalysis/package/MDAnalysis/lib/c_distances.o -lm -o build/lib.linux-x86_64-3.8/MDAnalysis/lib/c_distances.cpython-38-x86_64-linux-gnu.so
`

**Comment by IAlibay** [comment]
yeah I noticed that, the gcc calls seems to be related to the areas where openmp is getting picked up.

**Comment by IAlibay** [comment]
It looks like it might be down to our attempts to pick up OpenMP (so in the MacOS builds we build a serial version only), so something in that code path is ignoring the wishes of CC/CXX

**Comment by hmacdope** [comment]
Is openmp pulled from conda/pip or apt?

**Comment by IAlibay** [comment]
Here's an interesting one, looks like even when using gcc (in this case CC=gcc-7), you end up defaulting to "gcc" for the openmp sections: https://github.com/IAlibay/mdanalysis/runs/2109401928?check_suite_focus=true#step:8:119

**Comment by hmacdope** [comment]
Perhaps `fopenmp=libomp` will work.
